src/python/WMCore/WMRuntime/Watchdog.py

```diff
@@ -78,6 +81,46 @@ def setupMonitors(self, task, wmbsJob):
                 #This should be a config section
                 monitorArgs = getattr(task.data.watchdog, monitor)
                 args = monitorArgs.dictionary_()
+                # Scale resources according to the HTCondor runtime environment.
+                origCores = 1
+                for stepName in task.listAllStepNames():
+                    sh = task.getStepHelper(stepName)
+                    origCores = max(origCores, sh.getNumberOfCores())
+                resources = {'cores': origCores}
+                origMaxRSS = args.get('maxRSS')
+                if origMaxRSS:
+                    origMaxRSS = int(origMaxRSS/1000.) # HTCondor expects MB; we get KB.
+                    resources['memory'] = origMaxRSS
+                # Actually parses the HTCondor runtime
+                resizeResources(resources)
+                # We decided to only touch Watchdog settings if the number of cores changed.
+                # (even if this means the watchdog memory is wrong for a slot this size).
```

Comment:
Perhaps we should then change responsibilities on who controls the maxRSS? Instead of letting the assigner (Unified) set it, we just leave it to HTCondor. What do you think?